monitoring/logicmonitor.py

```diff
@@ -0,0 +1,2170 @@
+          action=sdt
+          duration=3000
+          starttime='2012-11-10 09:08'
+          company='{{ company }}'
+          user='{{ user }}'
+          password='{{ password }}'
+'''
+
+
+class LogicMonitor(object):
+
+    def __init__(self, module, **params):
+        self.__version__ = "1.0-python"
+        self.module = module
+        self.module.debug("Instantiating LogicMonitor object")
+
+        self.check_mode = False
+        self.company = params["company"]
+        self.user = params["user"]
+        self.password = params["password"]
+        self.fqdn = socket.getfqdn()
+        self.lm_url = "logicmonitor.com/santaba"
+        self.urlopen = open_url  # use the ansible provided open_url
+        self.__version__ = self.__version__ + "-ansible-module"
+
+    def rpc(self, action, params):
+        """Make a call to the LogicMonitor RPC library
+        and return the response"""
+        self.module.debug("Running LogicMonitor.rpc")
+
+        param_str = urllib.urlencode(params)
+        creds = urllib.urlencode(
+            {"c": self.company,
+                "u": self.user,
+                "p": self.password})
+
+        if param_str:
+            param_str = param_str + "&"
+
+        param_str = param_str + creds
+
+        try:
+            url = ("https://" + self.company + "." + self.lm_url +
+                   "/rpc/" + action + "?" + param_str)
+
+            # Set custom LogicMonitor header with version
+            headers = {"X-LM-User-Agent": self.__version__}
+
+            # Set headers
+            f = self.urlopen(url, headers=headers)
+
+            raw = f.read()
+            resp = json.loads(raw)
+            if resp["status"] == 403:
+                self.module.debug("Authentication failed.")
+                self.fail(msg="Error: " + resp["errmsg"])
+            else:
+                return raw
+        except IOError:
+            self.fail(msg="Error: Unknown exception making RPC call")
+
+    def do(self, action, params):
+        """Make a call to the LogicMonitor
+         server \"do\" function"""
+        self.module.debug("Running LogicMonitor.do...")
+
+        param_str = urllib.urlencode(params)
+        creds = (urllib.urlencode(
+            {"c": self.company,
+                "u": self.user,
+                "p": self.password}))
+
+        if param_str:
+            param_str = param_str + "&"
+        param_str = param_str + creds
+
+        try:
+            self.module.debug("Attempting to open URL: " +
+                              "https://" + self.company + "." + self.lm_url +
+                              "/do/" + action + "?" + param_str)
+            f = self.urlopen(
+                "https://" + self.company + "." + self.lm_url +
+                "/do/" + action + "?" + param_str)
+            return f.read()
+        except IOError:
+            # self.module.debug("Error opening URL. " + ioe)
+            self.fail("Unknown exception opening URL")
```

Comment:
Would like to see the IOError msgs and the url in the message. Though IOError at this point is unlikely to be url specific...